src/core/net/ip6.cpp

```diff
@@ -1037,6 +1037,19 @@ Error Ip6::PassToHost(OwnedPtr<Message> &aMessagePtr,
     }
 #endif
 
+#if OPENTHREAD_CONFIG_ANDROID_AOSP_ENABLE
+    // On Android the platform currently doesn't restrict link-local/mesh-local source
+    // addresses when forwarding multicast packets.
+    // For a multicast packet sent from link-local/mesh-local address to scope larger
+    // than realm-local, set the hop limit to 1 before sending to host, so this packet
+    // will not be forwarded by host.
+    if (aMessageInfo.GetSockAddr().IsMulticastLargerThanRealmLocal() &&
```

Comment:
Does it make sense to do the same for any multicast msg with ML/LL source? 
Not just `IsMulticastLargerThanRealmLocal`?

Reply:
The multicast forwarding logic on Android/other platforms would block msg to scope <= realm-local to be forwarded. Since this code modifies the original message (hoplimit), we may want to use it only when necessary?

Reply:
Make sense. Thanks.